### PR TITLE
[stdlib] Add nonce argument to TC account transactions

### PR DESCRIPTION
**Pull request by andll**
This diff adds nonce argument to following transaction scripts:

* freeze_account
* unfreeze_account
* burn
* tiered_mint

Also adding new script
* rotate_authentication_key_with_nonce

As a general rule nonce is added as a first argument

### CODE REVIEW

**Comment by andll** [comment]
I am not sure - what can I do with 'Move Prover test  did fail'?

**Comment by moeziniaf** [comment]
try running 'cargo run' in language/stdlib and and including generated files. otw it might be a code coverage issue

**Comment by moeziniaf** [comment]
also weirdly looks like a test is failing with the new nonce argument for mint

**Comment by andll** [comment]
>also weirdly looks like a test is failing with the new nonce argument for mint


Yeah, fixed that

**Comment by andll** [comment]
Ok, looks like tests are passing now.

File: language/stdlib/transaction_scripts/burn.move [hunk 1]
```diff
@@ -1,10 +1,12 @@
 script {
 use 0x0::Libra;
+use 0x0::SlidingNonce;
 
 // Permanently destroy the `Token`s stored in the oldest burn request under the `Preburn` resource.
 // This will only succeed if the sender has a `MintCapability<Token>` stored under their account, a
 // `Preburn<Token>` resource exists under `preburn_address`, and there is a pending burn request.
-fun main<Token>(preburn_address: address) {
+fun main<Token>(sliding_nonce: u64, preburn_address: address) {
```
Comment (sblackshear) [inline language/stdlib/transaction_scripts/burn.move:8]: Nit: add a comment about the role of the nonce here (and in the other scripts)?

**Pull request closed without merging**